pkg/config/security/api_client.go

```diff
@@ -0,0 +1,335 @@
+	}
+
+	vc := contextx.GetVerifyContext(ctx)
+	defaultErr := errorsx.G11NError("unable to create API client")
+	client := openapi.NewClientWithOptions(ctx, vc.Tenant, c.Client)
+
+	body, err := json.Marshal(apiClientConfig)
+	if err != nil {
+		vc.Logger.Errorf("Unable to marshal API client data; err=%v", err)
+		return "", defaultErr
+	}
+
+	response, err := client.CreateAPIClientWithBodyWithResponse(ctx, "application/json", bytes.NewBuffer(body), func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("Accept", "application/json")
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", vc.Token))
+		return nil
+	})
+	if err != nil {
+		vc.Logger.Errorf("Unable to create API client; err=%v", err)
+		return "", defaultErr
+	}
+
+	if response.StatusCode() != http.StatusCreated {
+		if err := errorsx.HandleCommonErrors(ctx, response.HTTPResponse, "unable to get API client"); err != nil {
+			vc.Logger.Errorf("unable to create the API client; err=%s", err.Error())
+			return "", err
+		}
+
+		vc.Logger.Errorf("unable to create the API client; code=%d, body=%s", response.StatusCode(), string(response.Body))
+		return "", defaultErr
+	}
+
+	// unmarshal the response body to get the ID
+	m := map[string]interface{}{}
+	resourceURI := ""
+	if err := json.Unmarshal(response.Body, &m); err != nil {
+		vc.Logger.Warnf("unable to unmarshal the response body to get the 'id'")
+		resourceURI = response.HTTPResponse.Header.Get("Location")
+	} else {
+		id := typesx.Map(m).SafeString("id", "")
+		resourceURI = fmt.Sprintf("%s/%s", response.HTTPResponse.Request.URL.String(), id)
+	}
+
+	return resourceURI, nil
+}
+
+func (c *ApiClient) GetAPIClient(ctx context.Context, clientName string) (*APIClientConfig, string, error) {
+	vc := contextx.GetVerifyContext(ctx)
+	client := openapi.NewClientWithOptions(ctx, vc.Tenant, c.Client)
+	id, err := c.GetAPIClientId(ctx, clientName)
+	if err != nil {
+		vc.Logger.Errorf("unable to get the group ID; err=%s", err.Error())
+		return nil, "", err
+	}
+
+	response, err := client.GetAPIClientWithResponse(ctx, id, func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("Accept", "application/json")
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", vc.Token))
+		return nil
+	})
+	if err != nil {
+		vc.Logger.Errorf("unable to get the API client; err=%s", err.Error())
+		return nil, "", err
+	}
+
+	if response.StatusCode() != http.StatusOK {
+		if err := errorsx.HandleCommonErrors(ctx, response.HTTPResponse, "unable to get API client"); err != nil {
+			vc.Logger.Errorf("unable to get the API client; err=%s", err.Error())
+			return nil, "", err
+		}
+
+		vc.Logger.Errorf("unable to get the API client; code=%d, body=%s", response.StatusCode(), string(response.Body))
+		return nil, "", errorsx.G11NError("unable to get the API client with clientName %s; status=%d", clientName, response.StatusCode())
+	}
+
+	Client := &APIClientConfig{}
```

Comment:
Should this not be `client` and not `Client`? Use `c` if it clashes. Pascal case is not used for variable names.